snow/engine/common/bootstrapper.go

```diff
@@ -116,17 +125,40 @@ func (b *Bootstrapper) AcceptedFrontier(validatorID ids.ShortID, requestID uint3
 	// Union the reported accepted frontier from [validatorID] with the accepted frontier we got from others
 	b.acceptedFrontier.Add(containerIDs...)
 
+	// still waiting on requests
+	if b.pendingAcceptedFrontier.Len() != 0 {
+		return nil
+	}
+
 	// We've received the accepted frontier from every bootstrap validator
 	// Ask each bootstrap validator to filter the list of containers that we were
 	// told are on the accepted frontier such that the list only contains containers
 	// they think are accepted
-	if b.pendingAcceptedFrontier.Len() == 0 {
-		vdrs := ids.ShortSet{}
-		vdrs.Union(b.pendingAccepted)
+	var err error
+	totalWeight := uint64(0)
+
+	// measure total weight
+	for _, beacon := range b.Beacons.List() {
+		if !b.failedAcceptedFrontierVdrs.Contains(beacon.ID()) {
+			totalWeight, err = math.Add64(totalWeight, beacon.Weight())
+			if err != nil {
```

Comment:
We should at least log an error here, and maybe even end the program, since this should never happen

Reply:
Yeah, I thought the same, I was following the lead from other places where we calculate the weight. 

I don't see a reason to `Fatal` but definitely would be an improvement to log these. 👌